sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -401,23 +576,52 @@ private boolean onChallengeAndClearCacheSync(KeyVaultCredentialPolicy policy, Ht
 
     private static class MutableTestCredential implements TokenCredential {
         private String credential;
+        private List<Function<TokenRequestContext, Boolean>> assertions;
 
-        public MutableTestCredential() {
+        public MutableTestCredential(List<Function<TokenRequestContext, Boolean>> assertions) {
             this.credential = new Random().toString();
+            this.assertions = assertions;
         }
 
         /**
-         * @throws RuntimeException If the UTF-8 encoding isn't supported.
+         * @throws RuntimeException if any of the assertions fail.
          */
         @Override
-        public Mono<AccessToken> getToken(TokenRequestContext request) {
-            if (request.isCaeEnabled() && request.getClaims() != null) {
+        public Mono<AccessToken> getToken(TokenRequestContext requestContext) {
+            if (requestContext.isCaeEnabled() && requestContext.getClaims() != null) {
+                assertEquals("{\"access_token\":{\"acrs\":{\"essential\":true,\"value\":\"cp1\"}}}",
```

Comment:
`BASE_ASSERTIONS` seems to cover this, do you need both? The condition makes this assertion look buggy